### DISCUSSION TITLE
Subscriptioon connection reset

### DISCUSSION THREAD

**Discussion by kisinga** [pocketbase/pocketbase#3338] [Q&A] [answered]
I'm using the javascript SDK V 0.18.0 on edge for Mac M2.
When I create a subscription to a resource, the connection gets reset after a few minutes of inactivity.
<img width="476" alt="image" src="https://github.com/pocketbase/pocketbase/assets/9964104/2022c555-e705-4d97-b3b1-b9124afda18e">

I suspect the issue may be related to [this discussion](https://github.com/pocketbase/pocketbase/discussions/2323) where @ganigeorgiev mentioned that this is normal but the SDK should re-establish a connection, but this is not happening on my end as I have to refresh the whole page to get the subscription to start working again.

**Comment by ganigeorgiev** [comment]
Are you able to reproduce the issue locally?

Do you have a reverse proxy?

Currently the idle disconnect timeout is set by default to 5mins.

The `ERR_CONNECTION_RESET` error I guess could indicate that the client wasn't able to connect to the PocketBase server for some reason but I have no idea what could be the cause of it.

You can try to checks the logs in the Admin UI to see if a connect request has even reached the server and/or start the executable with `--debug` and monitor the console output.

**Reply by ganigeorgiev** [reply, marked as answer]
I've just tested it locally and I'm not able to reproduce the error.

To rule out a PocketBase issue, you can try to create a static html file in `pb_public/index.html`:

```html
<!DOCTYPE html />
<html>
    <head>
        <meta charset="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <title>Test</title>
    </head>
    <body>
        <script type="module">
            import PocketBase from "https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@0.18.0/dist/pocketbase.es.mjs";

            const pb = new PocketBase("http://127.0.0.1:8090");

            await pb.collection("example").subscribe("*", (e) => {
                console.log(e)
            })
        </script>
    </body>
</html>
```

Navigate to `http://127.0.0.1:8090/`, open the browser's _"Inspect Tool > Network"_, and shortly after that restart (stop/start) the PocketBase executable.

Your Network tab should look something like (_I'm using Chrome 117_):

![network_tab_screenshot](https://i.imgur.com/p5NsJpo.png)

_(the red/error requests are the reconnect attempts while the PocketBase server was down)_

**Comment by kisinga** [comment]
@ganigeorgiev you're absolutely right. This error doesnt occur on localhost, indicating that It's probably my azure setup. I'll try debug it from there but since that's outside the scope of this discussion, i'll just close this
Thanks a bunch
